sympy/solvers/solveset.py

```diff
@@ -538,6 +538,7 @@ def _solve_as_rational(f, symbol, solveset_solver, as_poly_solver):
 def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
+    f_original = f
```

Comment:
It's no longer the original function at this point. I would move that line to before the trigsimp line.